### PR TITLE
Apple Pay Integration [JudoKit Invoker]

### PR DESCRIPTION
**Pull request by mpetrenco**
- Added method to the JudoAPI to invoke ApplePay payments;
- Added two new properties to the JPResponse object to save billing/shipping addresses;

**NOTE:**
For demo purposes, the JPResponse object has been mocked and the whole response/error logic has been skipped entirely. This is done to showcase that we successfully obtain the billing/shipping information, presenting it into the payment summary screen;

### CODE REVIEW

File: Source/JudoKit.m [hunk 1]
```diff
@@ -56,6 +60,13 @@ @interface JudoKit ()
 @property (nonatomic, strong) NSString *deviceIdentifier;
 @end
 
+@interface JudoKit ()
+@property (nonatomic, strong) ApplePayManager *manager;
```
Comment (zfoltin) [inline Source/JudoKit.m]: can these be moved up inside the previous `@interface JudoKit ()`

File: Source/JudoKit.m [hunk 2]
```diff
@@ -439,3 +450,111 @@ - (void)invokeTokenPreAuth:(NSString *)judoId
...
+
+@implementation JudoKit (ApplePay)
+
+#pragma mark - Apple Pay invocation method
+
+- (void)invokeApplePayWithConfiguration:(ApplePayConfiguration *)configuration
+                             completion:(JudoCompletionBlock)completion {
+    
+    self.configuration = configuration;
+    self.configuration.merchantId = @"merchant-com.judopay.JudoKitObjC";
```
Comment (zfoltin) [inline Source/JudoKit.m]: might not want to override this?

File: Source/JudoKit.m [hunk 3]
```diff
@@ -439,3 +450,111 @@ - (void)invokeTokenPreAuth:(NSString *)judoId
...
+    } else {
+        transaction = [self paymentWithJudoId:self.configuration.judoId
+                                       amount:self.manager.jpAmount
+                                    reference:self.manager.jpReference];
+    }
+    
+    NSError *error;
+    [transaction setPkPayment:payment error:&error];
+    
+    [transaction sendWithCompletion:^(JPResponse *response, NSError *error) {
```
Comment (zfoltin) [inline Source/JudoKit.m]: if there was an error in `setPkPayment` we might not want to send the transaction, right?

File: Source/JudoKit.m [hunk 4]
```diff
@@ -439,3 +450,111 @@ - (void)invokeTokenPreAuth:(NSString *)judoId
...
+        
+        //------------------------------------------------------------------------------
+        // TODO: Remove once finished showcasing billing / shipping contact information
+        //
+        // This is used for mocking purposes only to showcase the shipping
+        // information displayed to the merchant.
+        //-------------------------------------------------------------------------------
+        
+        response = self.mockJPResponse;
```
Comment (zfoltin) [inline Source/JudoKit.m]: Wouldn't it be better to use a compiler flag to drive this behaviour?

File: Source/JudoKit.m (hunk 2 above)
Comment (mpetrenco) [reply Source/JudoKit.m]: Oh wow.
Completely missed that.

**Pull request merged**